Feat: x-pack cloud id/auth for monitoring/management (#11496)
elastic/logstash: docs/static/settings/configuration-management-settings.asciidoc

`xpack.management.elasticsearch.ssl.keystore.password`::

Optional setting that provides the password to the keystore.

`xpack.management.elasticsearch.cloud_id`::

> If you're using Elasticsearch cloud, you should specify the identifier here.

Reviewer: I don't _think_ we refer to it as `Elasticsearch cloud`, but rather as `Elastic Cloud`, so maybe:

```suggestion
If you're using {es} in Elastic Cloud, you should specify the identifier here.
```

Reviewer 2: You're on the right track!  Should be `{es} in {ecloud}`
I'm doing a quick local test to be sure that renders expected results.

Reviewer 2: ^^ We have attributes in place that sub in the correct product names.  It's good for ensuring consistency and handy in case we rename or rebrand.

This is an alternative to specifying `xpack.management.elasticsearch.hosts`,

> which should not be used if `cloud_id` is configured.

Reviewer 2: This setting is an alternative to `xpack.management.elasticsearch.hosts`.
If `cloud_id` is configured, `xpack.management.elasticsearch.hosts` should not be used.

> if `cloud_id` is set having also a `hosts` configuration makes no sense.

Reviewer 3: I might consider rewording this to something like "If `cloud_id` is set, `hosts` must not be configured. @karenzone - thoughts?

> if `cloud_id` is set having also a `hosts` configuration makes no sense.

Reviewer: Wording is a bit awkward. So is this, but maybe less so?

```suggestion
which should not be used if `cloud_id` is configured.
```

This {es} instance will store the Logstash pipeline configurations and metadata.

`xpack.management.elasticsearch.cloud_auth`::

> If you're using Elasticsearch cloud, you can set your auth credentials here.

Reviewer 2: `{es} in {ecloud}

This is an alternative to having `xpack.management.elasticsearch.username`
and `xpack.management.elasticsearch.password` - those should not be used if

> `cloud_auth` is configured.

Reviewer 2: Suggested wording:

This setting is an alternative to both `xpack.management.elasticsearch.username`
and `xpack.management.elasticsearch.password`. If`cloud_auth` is configured,
those settings should not be used.
